### DISCUSSION TITLE
Works for JK_BD6A20S12P

### DISCUSSION THREAD

**Discussion by asiridissa** [syssi/esphome-jk-bms#339] [Show and tell]
Works very well for BD6A20S12P with ESP8266 D1 Mini.
Used the GPS(Default RS485) port.
It is awesome with Home Assistant server.
I need to automate balancing switch. So planning on upgrade to ESP32 BLE.

Yaml
[jk-bms Working.txt](https://github.com/syssi/esphome-jk-bms/files/11713927/jk-bms.Working.txt)

BMS Info
![Screenshot_20230604_173030](https://github.com/syssi/esphome-jk-bms/assets/3539388/f664a46c-edbf-401b-804e-f5367c181ac8)

Home Assistant Dashboard
![image](https://github.com/syssi/esphome-jk-bms/assets/3539388/ef16d812-ade5-4772-9d06-6770be68cc5b)

Thanks @syssi and all the contributors.

**Comment by YuZhengshun** [comment]
May I ask, did you configure it as 000 4G-GPS, 001 RS485, or another option?
Thank you very much.

**Reply by syssi** [reply]
If you want to use the `4G-GPS` protocol you have to use the `jk_bms` component: [esp32-example.yaml](https://github.com/syssi/esphome-jk-bms/blob/main/esp32-example.yaml)

If you want to use the `001 RS485` protocol (if available!) you have to use the `esp32-jk-pb-modbus-example.yaml`.
